src/clipboard_handlers/tables_clipboard.ts

```diff
@@ -69,8 +69,16 @@ export class TableClipboardHandler extends AbstractCellClipboardHandler<
           zones.some((z) => isZoneInside(tableZone, z))
         ) {
           copiedTablesIds.add(table.id);
+          let { numberOfRows } = zoneToDimension(tableZone);
+          for (let r = tableZone.top; r <= tableZone.bottom; r++) {
+            if (!isCutOperation && !rowsIndexes.includes(r)) {
+              numberOfRows--;
+            }
+          }
+          const range = this.getters.getRangeData(coreTable.range);
+          range._zone = { ...range._zone, bottom: range._zone.top + numberOfRows - 1 };
```

Comment:
avoid using and manipulating `_zone`. The `_` indicates it's (almost) private and shouldn't be used.
You can use `getters.extendRange` to reduce the range.

```ts
          const range = coreTable.range;
          const newRange = this.getters.extendRange(coreTable.range, "ROW", range.zone.bottom - range.zone.top - numberOfRows + 1)
          copiedTable = {
            range: this.getters.getRangeData(newRange),
```